core/src/main/java/nl/nn/adapterframework/http/HttpSender.java

```diff
@@ -438,33 +436,26 @@ protected HttpEntity createMultiPartEntity(String message, ParameterValueList pa
 			if (log.isDebugEnabled()) log.debug(getLogPrefix()+"appended stringpart ["+getFirstBodyPartName()+"] with value ["+message+"]");
 		}
 		if (parameters!=null) {
-			for(int i=0; i<parameters.size(); i++) {
-				ParameterValue pv = parameters.getParameterValue(i);
-				ParameterType paramType = pv.getDefinition().getType();
+			for(ParameterValue pv : parameters) {
 				String name = pv.getDefinition().getName();
 
 				// Skip parameters that are configured as ignored
 				if (skipParameter(name))
 					continue;
 
-
-				if (paramType == ParameterType.INPUTSTREAM) {
-					Object value = pv.getValue();
-					if (value instanceof InputStream) {
-						InputStream fis = (InputStream)value;
-						String fileName = null;
-						String sessionKey = pv.getDefinition().getSessionKey();
-						if (sessionKey != null) {
-							fileName = session.getMessage(sessionKey + "Name").asString();
-						}
-
-						entity.addPart(createMultipartBodypart(name, fis, fileName));
-						if (log.isDebugEnabled()) log.debug(getLogPrefix()+"appended filepart ["+name+"] with value ["+value+"] and name ["+fileName+"]");
-					} else {
-						throw new SenderException(getLogPrefix()+"unknown inputstream ["+value.getClass()+"] for parameter ["+name+"]");
+				Message msg = pv.asMessage();
+				if (msg.isBinary()) {
```

Comment:
Je kan niet zomaar zeggen dat als een Message binary is, dat het attachment dat ook moet zijn.
De check op het paramtertype is nu verdwenen

Reply:
hmm